src/include/configuration/configuration_manager.h

```diff
@@ -0,0 +1,133 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// configuration_manager.h
+//
+// Identification: src/include/configuration/configuration_manager.h
+//
+// Copyright (c) 2015-2017, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+#pragma once
+
+#include "type/types.h"
+#include "common/exception.h"
+#include "common/printable.h"
+
+namespace peloton {
+namespace configuration {
+  class Config;
+}}
+using peloton::configuration::Config;
+
+namespace peloton {
+namespace configuration {
+
+class Config {
+public:
+  static uint64_t GET_INT(const std::string& name);
+  static bool GET_BOOL(const std::string& name);
+  static std::string GET_STRING(const std::string& name);
+
+  static void SET_INT(const std::string &name, uint64_t value);
+  static void SET_BOOL(const std::string &name, bool value);
+  static void SET_STRING(const std::string &name, const std::string &value);
+
+  static void init_parameters();
+  static void drop_parameters();
+};
+
+class ConfigurationManager : public Printable {
+public:
+  static ConfigurationManager* GetInstance();
+
+  void InitializeCatalog();
+
+  const std::string GetInfo() const;
+
+  void Clear();
+
+  void Display();
+
+  template<typename T>
+  T GetValue(const std::string &name) {
+    auto param = config.find(name);
+    if (param == config.end()) {
+      throw new ConfigurationException("no such configuration: " + name);
+    }
+    return to_value<T>(param->second.value);
+  }
+
+  template<typename T>
+  void SetValue(const std::string &name, const T &value) {
```

Comment:
We may want to switch this to use our own `Value` type object. Your casting is unsafe.